### PR TITLE
Add PHPStan to test environment with `max` level

### PR DESCRIPTION
**Pull request by clue**
This changeset adds PHPStan on `max` level to the test environment for all supported PHP versions as discussed in #56. It runs the maximum level supported with all errors addressed without having to resort to a baseline. The changeset size seems reasonable and this does not other affect our public API, so this should be safe to apply.

Builds on top of #60, #59, #58, #44 and https://github.com/reactphp/async/pull/76

### CODE REVIEW

**Review by SimonFrings — APPROVED** [review]
Nice :+1:

File: src/CacheInterface.php [hunk 1]
```diff
@@ -120,7 +120,7 @@ public function delete(string $key): PromiseInterface;
      *
      * @param string[] $keys A list of keys that can obtained in a single operation.
      * @param mixed $default Default value to return for keys that do not exist.
-     * @return PromiseInterface<array> Returns a promise which resolves to an `array` of cached values
+     * @return PromiseInterface<array<string,mixed>> Returns a promise which resolves to an `array` of cached values
```
Comment (WyriHaximus) [inline src/CacheInterface.php:123]: Have we considered making the cache typeable? (Out of scope for this PR of course.)

```suggestion
     * @return PromiseInterface<array<string,T>> Returns a promise which resolves to an `array` of cached values
```

Comment (clue) [reply src/CacheInterface.php:123]: Not sure how useful this is, but happy to discuss in a separate PR if this is something you want to look into :+1:

Comment (WyriHaximus) [reply src/CacheInterface.php:123]: I will do a PR once this is in together with one for DNS to show that off after doing some more research :+1: .

**Review by WyriHaximus — APPROVED** [review]

**Pull request merged**